diff --git a/dist/changesets-assemble-release-plan.cjs.js b/dist/changesets-assemble-release-plan.cjs.js
index ee5c0f67fabadeb112e9f238d8b144a4d125830f..07b659be617f32e70a9b484c668aa673e213f50e 100644
--- a/dist/changesets-assemble-release-plan.cjs.js
+++ b/dist/changesets-assemble-release-plan.cjs.js
@@ -388,7 +388,7 @@ function applyLinks(releases, packagesByName, linked) {
 
 function getPreVersion(version) {
   let parsed = semverParse__default["default"](version);
-  let preVersion = parsed.prerelease[1] === undefined ? -1 : parsed.prerelease[1];
+  let preVersion = parsed?.prerelease?.[1] === undefined ? -1 : parsed?.prerelease?.[1];
 
   if (typeof preVersion !== "number") {
     throw new errors.InternalError("preVersion is not a number");
diff --git a/dist/changesets-assemble-release-plan.esm.js b/dist/changesets-assemble-release-plan.esm.js
index bf5202626a164a7780650d333983c3479b078689..4a018d38b76a07c92eadae0bcec6d80bdb3595e8 100644
--- a/dist/changesets-assemble-release-plan.esm.js
+++ b/dist/changesets-assemble-release-plan.esm.js
@@ -377,7 +377,7 @@ function applyLinks(releases, packagesByName, linked) {
 
 function getPreVersion(version) {
   let parsed = semverParse(version);
-  let preVersion = parsed.prerelease[1] === undefined ? -1 : parsed.prerelease[1];
+  let preVersion = parsed?.prerelease?.[1] === undefined ? -1 : parsed?.prerelease?.[1];
 
   if (typeof preVersion !== "number") {
     throw new InternalError("preVersion is not a number");
diff --git a/src/index.ts b/src/index.ts
index c2808fcd430346759975a40e951d49142f4896b9..6861ebb8a6de6cbb85839428558ac499f2c507a8 100644
--- a/src/index.ts
+++ b/src/index.ts
@@ -24,7 +24,7 @@ type SnapshotReleaseParameters = {
 function getPreVersion(version: string) {
   let parsed = semverParse(version)!;
   let preVersion =
-    parsed.prerelease[1] === undefined ? -1 : parsed.prerelease[1];
+    parsed?.prerelease?.[1] === undefined ? -1 : parsed?.prerelease?.[1];
   if (typeof preVersion !== "number") {
     throw new InternalError("preVersion is not a number");
   }
