diff --git a/dist/changesets-cli.cjs.js b/dist/changesets-cli.cjs.js
index 02083f394e874d25ca5ec9e7faaab83fe06e3761..cfd543d0293611a31bf6cc61d1f95f49b9e4bd07 100644
--- a/dist/changesets-cli.cjs.js
+++ b/dist/changesets-cli.cjs.js
@@ -793,7 +793,11 @@ let getOtpCode = async twoFactorState => {
 
 async function internalPublish(pkgName, opts, twoFactorState) {
   let publishTool = await getPublishTool(opts.cwd);
-  let publishFlags = opts.access ? ["--access", opts.access] : [];
+  let publishFlags = publishTool.name !== "yarn" ? ["--json"] : [];
+
+  if (opts.access) {
+    publishFlags.push("--access", opts.access);
+  }
   publishFlags.push("--tag", opts.tag);
 
   if ((await twoFactorState.isRequired) && !ciInfo.isCI) {
diff --git a/dist/changesets-cli.esm.js b/dist/changesets-cli.esm.js
index 6b0c2cecd33c4ed89bfc964daf3ca5d1e482c1db..ab99d2ab534bf19c66d3ed47f66cbaaa22b63472 100644
--- a/dist/changesets-cli.esm.js
+++ b/dist/changesets-cli.esm.js
@@ -752,7 +752,11 @@ let getOtpCode = async twoFactorState => {
 
 async function internalPublish(pkgName, opts, twoFactorState) {
   let publishTool = await getPublishTool(opts.cwd);
-  let publishFlags = opts.access ? ["--access", opts.access] : [];
+  let publishFlags = publishTool.name !== "yarn" ? ["--json"] : [];
+
+  if (opts.access) {
+    publishFlags.push("--access", opts.access);
+  }
   publishFlags.push("--tag", opts.tag);
 
   if ((await twoFactorState.isRequired) && !isCI) {
